src/script/lua_api/l_object.cpp

```diff
@@ -433,10 +423,10 @@ int ObjectRef::l_set_local_animation(lua_State *L)
 
 	v2s32 frames[4];
 	for (int i=0;i<4;i++) {
-		if (!lua_isnil(L, 2+1))
+		if (!lua_isnoneornil(L, 2+1))
```

Comment:
API documentation needs to be adjusted accordingly to state that the argument is now entirely optional, not just "nilable"
-> `set_local_animation([idle, walk, dig, walk_while_dig, frame_speed])`

IMO this change is not necessary because the animation defaults (`0,0`) don't make sense in most cases.